### PR TITLE
Improve calculateRetryWaitMilliseconds

### PR DESCRIPTION
**Pull request by Dimon4eg**

### CODE REVIEW

File: ixwebsocket/IXWebSocket.cpp [hunk 1]
```diff
@@ -13,16 +13,21 @@
...
-        // and we'll be constantly retrying to connect.
-        uint64_t wait_time = ((uint64_t) std::pow(2, retry_count) * 100L);
+        uint64_t wait_time;
 
-        // cap the wait time to 10s, or to retry_count == 10 for which wait_time > 10s
-        uint64_t tenSeconds = 10 * 1000;
-        return (wait_time > tenSeconds || retry_count > 10) ? tenSeconds : wait_time;
+        if (retry_count <= 6)
+        {
+            // max wait_time is 6400
```
Comment (bsergean) [inline ixwebsocket/IXWebSocket.cpp]: max wait_time is 6400ms

File: ixwebsocket/IXWebSocket.cpp [hunk 2]
```diff
@@ -13,16 +13,21 @@
 
 namespace
 {
-    uint64_t calculateRetryWaitMilliseconds(uint64_t retry_count)
+    uint64_t calculateRetryWaitMilliseconds(uint32_t retry_count)
     {
-        // This will overflow quite fast for large value of retry_count
```
Comment (bsergean) [inline ixwebsocket/IXWebSocket.cpp]: Can you keep that comment about overflow for the next programmer who come here ?

Comment (Dimon4eg) [reply ixwebsocket/IXWebSocket.cpp]: No overflow now possible as I'm checking before: `if (retry_count <= 6)`

File: ixwebsocket/IXWebSocket.cpp (hunk 1 above)
Comment (Dimon4eg) [reply ixwebsocket/IXWebSocket.cpp]: That comments no more correct or am I wrong?

Comment (bsergean) [reply ixwebsocket/IXWebSocket.cpp]: Add 'ms'.

File: ixwebsocket/IXWebSocket.cpp (hunk 2 above)
Comment (bsergean) [reply ixwebsocket/IXWebSocket.cpp]: Sure, but we should explain why we have this kind of odd magic value (6). Normally we wouldn't have to do that and would just use the pow(2) thing.

**Pull request merged**